### PR TITLE
add temporal head and MPJPELoss

### PR DESCRIPTION
**Pull request by cherryjm**
- add temporal head
- add MPJPELoss for 3D  pose estimation

### CODE REVIEW

**Review by jin-s13 — APPROVED** [review]
It looks good to me.

File: mmpose/core/post_processing/post_transforms.py [hunk 1]
```diff
@@ -75,6 +75,32 @@ def fliplr_regression(regression, flip_pairs):
...
+
+        - regression_flipped (np.ndarray([N, K, 3])): Flipped joints.
+    """
+    regression_flipped = regression.copy()
+    # Swap left-right parts
+    for left, right in flip_pairs:
+        regression_flipped[:, left, :] = regression[:, right, :]
+        regression_flipped[:, right, :] = regression[:, left, :]
+    # Flip horizontally
+    regression_flipped[:, :, 0] = -regression_flipped[:, :, 0]
```
Comment (innerlee) [inline mmpose/core/post_processing/post_transforms.py:100]: Double check this transformation to make sure that there are no issues concerning the absolute coordinates of pelvis

**Comment by innerlee** [comment]
The code is clean. An important thing is to make sure that the flip op correctly changes all affected coordinates.

File: mmpose/models/keypoint_heads/temporal_head.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+from mmcv.cnn import build_conv_layer, normal_init
+
+from mmpose.core.evaluation import compute_similarity_transform
+from mmpose.core.post_processing import fliplr_regression_3d
+from mmpose.models.builder import build_loss
+from mmpose.models.registry import HEADS
+
+
+@HEADS.register_module()
+class TemporalHead(nn.Module):
```
Comment (ly015) [inline mmpose/models/keypoint_heads/temporal_head.py:12]: Would it be better to use a more specific name, like TemporalRegressionHead or something?

Comment (cherryjm) [reply mmpose/models/keypoint_heads/temporal_head.py:12]: Yes, I was not sure about the name. Do I also need to change the file name?

Comment (innerlee) [reply mmpose/models/keypoint_heads/temporal_head.py:12]: TemporalRegressionHead looks good. The filename should be changed accordingly

File: mmpose/models/keypoint_heads/temporal_regression_head.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        """
+        output = self.forward(x)
+
+        if flip_pairs is not None:
+            output_regression = fliplr_regression_3d(
+                output.detach().cpu().numpy(), flip_pairs)
+        else:
+            output_regression = output.detach().cpu().numpy()
+        return output_regression
+
```
Comment (ly015) [inline mmpose/models/keypoint_heads/temporal_regression_head.py:127]: I think we need a "decode" method for this head. Please refer to other keypoint heads.

File: mmpose/models/keypoint_heads/temporal_regression_head.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    def decode(self, metas, output):
+        """Decode the keypoints from output regression.
+
+        Args:
+            metas (list(dict)): Information about data augmentation.
+                By default this includes:
+                - "target_image_path": path to the image file
+            output (np.ndarray[N, K, 3]): predicted regression vector.
+        """
+        target_image_paths = []
```
Comment (innerlee) [inline mmpose/models/keypoint_heads/temporal_regression_head.py:137]: Save a few lines by

```python
target_image_paths = [m['target_image_path'] for m in metas]
```

File: mmpose/core/post_processing/post_transforms.py (hunk 1 above)
Comment (innerlee) [reply mmpose/core/post_processing/post_transforms.py:100]: Confirmed in person that this is intended.

**Pull request merged**